crates/polars-lazy/src/dsl/functions.rs

```diff
@@ -118,12 +118,11 @@ pub(crate) fn concat_impl<L: AsRef<[LazyFrame]>>(
 #[cfg(feature = "diagonal_concat")]
 /// Concat [LazyFrame]s diagonally.
 /// Calls [`concat`][concat()] internally.
-pub fn diag_concat_lf<L: AsRef<[LazyFrame]>>(
-    lfs: L,
-    rechunk: bool,
-    parallel: bool,
+pub fn concat_lf_diagonal<L: AsRef<[LazyFrame]>>(
+    inputs: L,
+    args: UnionArgs,
 ) -> PolarsResult<LazyFrame> {
-    let lfs = lfs.as_ref().to_vec();
+    let lfs = inputs.as_ref().to_vec();
```

Comment:
I see this wasn't introduced here, but let's fix it now. 

We don't need to allocate a Vec. We can clone the `LazyFrame` upon iteration. This saves the heap allocation of the vec with pointers to the logical plan.

Reply:
Done ✌️:sunglasses: